docs/multihost-networking-tutorial.md

```diff
@@ -0,0 +1,254 @@
+[menu.main]
+parent="workw_swarm"
+weight=3
++++
+<![end-metadata]-->
+
+# Multi-host networking deployment tutorial
+This step-by-step tutorial aims to walk you through a full Docker Swarm deployment with all the containers sharing a same LAN. Tutorial [first published here](https://www.auzias.net/en/docker-network-multihost/).
+
+## Prerequisites
+### Environment
+This tutorial has been tested with:
+  - docker engine 1.10.2.
+  - Swarm server 1.1.3.
+  - [Consul 0.6.3](http://consul.io/), needed for discovery for the [Swarm containers](https://docs.docker.com/swarm/discovery/).
+
+### Discovery
+Both Swarm agents and containers require a discovery mean; the former is needed to create the cluster itself, while the latter enables the multi-host networking. Note that the container discovery is not necessary if the Swarm containers do not need a common LAN.
+
+#### Containers discovery
+[Consul](http://consul.io/) is used for the multi-host networking discovery, any other key-value store can be used -- note that the engine supports [Consul](http://consul.io/), [etcd]https://coreos.com/etcd/, and [ZooKeeper](https://zookeeper.apache.org/). If you don't have yet any affinities you might want to read this [comparison post](http://technologyconversations.com/2015/09/08/service-discovery-zookeeper-vs-etcd-vs-consul/).
+
+#### Swarm agents discovery
+Static file or token can be used for the Swarm agent discovery. Tokens use REST API, and thus the network and an external server. The static file is preferred as it is faster, safer and also easier. Both methods are detailed below.
+
+### Docker daemon
+All nodes should run the Docker daemon as follow:
+
+    /usr/bin/docker daemon -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock --cluster-advertise eth0:2375 --cluster-store consul://127.0.0.1:8500
+
+#### Options details:
+
+    -H tcp://0.0.0.0:2375
+
+Binds the daemon to an interface to allow be part of the [swarm](https://docs.docker.com/swarm/) cluster. An IP address can obviously be specified, it is a better solution if you have several NIC. The Docker daemons bind on the port 2375, the Swarm daemon will bind on the port 5732, see below.
+
+    --cluster-advertise eth0:2375
+
+Defines the interface and the port of the docker daemon should use to advertise itself.
+
+    --cluster-store consul://127.0.0.1:8500
+
+Defines the URI of the distributed kv storage backend. As consul is distributed the node can
+
+As [consul](http://consul.io/) is distributed a local endpoint is used instead of the consul master. Hence the master can be changed without having to restart Docker daemon on all Swarm agents and the daemons communicate locally with the cluster-storage while the cluster-storages communicate together through the network. It also allows to select the Swarm manager after the daemon has been started.
+
+### The sh-aliases used
+In the following commands these two aliases are used:
+
+    alias ldocker='docker -H tcp://0.0.0.0:2375'
```

Comment:
I do not think adding a command alias is a good way.
We can just achieve this by `export DOCKER_HOST=tcp://0.0.0.0:2375`.